consensus/storage_fault_slasher.go

```diff
@@ -0,0 +1,90 @@
+package consensus
```

Comment:
I would say this *definitely* shouldn't be part of consensus. If you weren't about to land this, I would say blocking. Please follow up with a move to somewhere like `protocol/storage`.